src/sections/create-dataset/CreateDatasetProvider.tsx

```diff
@@ -0,0 +1,38 @@
+import { PropsWithChildren, useState, ReactNode } from 'react'
+import { useTranslation } from 'react-i18next'
+import styles from '/src/sections/dataset/Dataset.module.scss'
+import { SeparationLine } from '../../components/layout/SeparationLine/SeparationLine'
+import { FormContext } from './CreateDatasetContext'
+import { CreateDatasetFormFields } from '../../dataset/domain/useCases/createDataset'
+// Add a type for the children prop
+type FormProviderProps = {
+  children: ReactNode
+}
+// Context Provider Component
+export const CreateDatasetProvider: React.FC<FormProviderProps> = ({
```

Comment:
Uncouple this component from the provider, and just call the CreateDatasetForm. Provider and Context are being merged into the hook in the context file.